### PR TITLE
Corrections to migration guide

### PR DESCRIPTION
**Pull request by zuiderkwast**
- User the term "Redis OSS" and "open source Redis", rather than "Redis Community Edition". The latter refers to the source-available versions from 7.4 and later.
- Mention that Valkey is backward compatible with all old Redis versions.
- Update the migration table to include Redis OSS 2.x - 7.2.x. Users have been confused by the table that seems to indicate that it's not possible to upgrade from Redis 6.2 to Valkey 8.0, but it is.
- Mention that it may be possible to migrate from proprietary Redis, but that it's not covered by this document. (For example, writing the data to disk using AOF without preamble and loading in Valkey can probably work.)

### CODE REVIEW

**Review by nastena1606 — APPROVED** [review]
LGTM

File: topics/migration.md [hunk 1]
```diff
@@ -18,15 +18,19 @@ Refer to [install Valkey](installation.md) for installation options.
 
 ### Migration compatibility matrix
 
-You can migrate Redis server to Valkey. 
-The following table provides migration options depending on the Redis version you run:
+You can migrate a Redis server to Valkey.
+Valkey is backward compatible with Redis OSS 7.2 and all earlier open source Redis versions.
```
Comment (madolson) [inline topics/migration.md:22]:
```suggestion
Valkey is compatible with Redis OSS 7.2 and all earlier open source Redis versions.
```
It's a different engine, it's not backwards compatibility.

File: topics/migration.md [hunk 2]
```diff
@@ -18,15 +18,19 @@ Refer to [install Valkey](installation.md) for installation options.
...
+Migrating from any open source Redis version to Valkey is just an upgrade.
+Redis 7.4 and later are not open source and the data files are no longer compatible with Valkey.
```
Comment (madolson) [inline topics/migration.md:24]:
```suggestion
Redis Community Edition (CE), versions 7.4 and later, are not open source and the data files are not compatible with Valkey.
```

File: topics/migration.md [hunk 3]
```diff
@@ -18,15 +18,19 @@ Refer to [install Valkey](installation.md) for installation options.
...
-| 7.2.x | 7.2.x  |
-| 7.2.x | 8.0.x  |
-| 7.4   | n/a    |
+The following table provides migration options depending on the open source Redis version you run:
+
+| Redis                 | Valkey |
+|-----------------------|--------|
+| 2.x - 7.2.x           | 7.2.x  |
+| 2.x - 7.2.x           | 8.0    |
+| 7.4 (not open source) | n/a    |
```
Comment (madolson) [inline topics/migration.md:33]:
```suggestion
The following table provides migration options depending on the Redis version you run:

| Redis                 | Valkey |
|-----------------------|--------|
| OSS 2.x - 7.2.x       | 7.2.x  |
| OSS 2.x - 7.2.x       | 8.0    |
| CE 7.4                | n/a    |
```

I found it odd that you added open source in the header, but then later called it non-OSS. Maybe this is clearer?

File: topics/migration.md [hunk 4]
```diff
@@ -18,15 +18,19 @@ Refer to [install Valkey](installation.md) for installation options.
 
 ### Migration compatibility matrix
 
-You can migrate Redis server to Valkey. 
+You can migrate a Redis server to Valkey.
+Valkey is compatible with Redis OSS 7.2 and all earlier open source Redis versions.
+Migrating from any open source Redis version to Valkey is just an upgrade.
```
Comment (stockholmux) [inline topics/migration.md:23]: I would change 'just' -> 'effectively'

File: topics/migration.md [hunk 5]
```diff
@@ -18,15 +18,19 @@ Refer to [install Valkey](installation.md) for installation options.
...
+Redis Community Edition (CE), versions 7.4 and later, are not open source and the data files are not compatible with Valkey.
```
Comment (stockholmux) [inline topics/migration.md:24]: I assume you mean RDB files? I don't think we use the term 'data files' often. If there is a link that explains the difference between 7.4 and Valkey that would be helpf to the reader.

**Review by stockholmux — CHANGES_REQUESTED** [review]

File: topics/migration.md (hunk 5 above)
Comment (madolson) [reply topics/migration.md:24]: It would also include AOF files I presume, so not just RDB.

**Review by madolson — APPROVED** [review]
I'm OK with it as is, but think Kyle's suggestions also make sense.

File: topics/migration.md (hunk 4 above)
Comment (zuiderkwast) [reply topics/migration.md:23]:
```suggestion
Migrating from any open source Redis version to Valkey is effectively an upgrade.
```

File: topics/migration.md (hunk 5 above)
Comment (zuiderkwast) [reply topics/migration.md:24]: I don't have time to go into the details how you can migrate. It's possible to use various tools, like rdblib and RediShake, that can convert between different RDB versions, connect as a replica and emit commands to a Valkey node, or something like that. To do this without any service impact is probably possible, but I don't have enough experience with it.

I just want to remove the parts that are incorrect in this document, and to make it not imply that it's impossible to migrate between any other versions than those listed in the table.

Comment (zuiderkwast) [reply topics/migration.md:24]: Btw, Kyle, I agree with your comments. It's just a matter of prioritization.

**Pull request merged**